### PR TITLE
MAT-6659: Implement partial routing

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-6659](https://jira.cms.gov/browse/MAT-6659)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is into the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qdm/EditTestCase.tsx [hunk 1]
```diff
@@ -186,6 +186,7 @@ const EditTestCase = () => {
           setCurrentTestCase(nextTc);
         })
         .catch((error) => {
+          // console.log("error", error);
```
Comment (chubert-sb) [inline src/components/editTestCase/qdm/EditTestCase.tsx]: could probably remove this commented line

File: src/components/testCaseLanding/common/TestCaseListSideBarNav.test.tsx [hunk 2]
```diff
@@ -76,12 +74,12 @@ describe("TestCase component", () => {
...
     userEvent.click(activeLink);
-    expect(onChange).toHaveBeenCalled();
+    // expect(onChange).toHaveBeenCalled();
     const inactiveLink = screen.getByRole("tab", {
       name: "Population Criteria 1",
     });
     expect(inactiveLink).toBeInTheDocument();
     userEvent.click(inactiveLink);
-    expect(onChange).toHaveBeenCalledTimes(2);
+    // expect(onChange).toHaveBeenCalledTimes(2);
```
Comment (chubert-sb) [inline src/components/testCaseLanding/common/TestCaseListSideBarNav.test.tsx]: Commented lines here too

**Review by chubert-sb — DISMISSED** [review]

**Review by sb-prateekkeerthi — APPROVED** [review]
